### DISCUSSION TITLE
Should compression be an active feature of grass?

### DISCUSSION THREAD

**Discussion by tristan-f-r** [ConnorSkees/grass#69] [Ideas]
Parcel has an amazing library called [lightningcss](https://lightningcss.dev/) which offers better performance than both CSSNano and esbuild's css minifier. Given that there is already a minification tool for CSS, should grass also offer minification services that make grass's minification features obsolete. Given this, should grass _still_ actively support minification?

**Comment by connorskees** [comment]
Yes, this project should and will always support minification. If users want particular a minification pass that `grass` does not do, they should either use a tool that implements that pass as a post-processing step or open a feature request.

It doesn't make sense to remove minification from this library because there exist other tools that do CSS compression. Emitting compressed output is extremely cheap for us and should satisfy the vast majority of use cases.

**Reply by tristan-f-r** [reply]
Actually, I think this came more from a place of confusion - I originally assumed that dart-sass doesn't have any minification support, but [dart-sass *does* have a minification feature on their CLI](https://sass-lang.com/documentation/cli/dart-sass). However, the code that internally implements minification inside grass already covers some small optimization cases (like removing trailing zeros and optimizing colors).

Instead of removing the minification feature entirely, should this be handed off *to* lightningcss to offer compatibility with node-sass's minification feature, instead of creating an in-house solution for minification that may get more and more features over time?

(I do very doubt the fact that trailing zeros and color optimization will be the only things that will be added to the minification system)

**Reply by connorskees** [reply]
Ah. Yes, the other issue is that `grass` must implement `dart-sass` compatible compressed output to comply with the official spec tests. Sass as a language is limited in what optimizations it can do, since it explicitly does not know about specific properties, so it can never e.g. emit shorthand syntax.

We can't by-default use `lightningcss` to compress output, but a 3rd-party tool could combine the two if it chose to do so.

Sass does a lot more than just removing trailing zeroes and optimizing colors. If there are optimizations you think are missing, you should open a feature request either here or on the [official spec repo](https://github.com/sass/sass).

**Reply by tristan-f-r** [reply]
> Yes, the other issue is that `grass` must implement `dart-sass` compatible compressed output to comply with the official spec tests.

That makes a lot more sense, thanks for clarifying!
